app/knight.py

```diff
@@ -0,0 +1,35 @@
+from equiption.armour import Armour
+from equiption.potion import Potion
+from equiption.weapon import Weapon
+
+
+class Knight:
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list = None,
+                 weapon: dict = None,
+                 potion: dict = None) -> None:
+
+        self.name = name
+
+        self.weapon = Weapon(weapon["name"], weapon["power"])
```

Comment:
The `weapon` parameter is expected to be a dictionary with keys 'name' and 'power'. Ensure that this structure is maintained when passing data to the `Weapon` class constructor .